# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDProcessManager.h
# Commit: bf32f5089547
# Full Hash: bf32f5089547afa36393ab6771b45e7ab5ba61ef
# Author: alwu <alwu@mozilla.com>
# Date: 2021-03-03 04:16:49
# Description:
#   Bug 1685666 - check RDDChild before creating content bridge. r=mattwoodrow
#   
#   As the process of creating content process is async, the RDD process might be shutdown before that (eg. crash). So we should check if RDD still exists first.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106827
# ==============================================================================

diff -r 55b0990074bc -r bf32f5089547 dom/media/ipc/RDDProcessManager.h
--- a/dom/media/ipc/RDDProcessManager.h	Tue Mar 02 04:36:57 2021 +0000
+++ b/dom/media/ipc/RDDProcessManager.h	Tue Mar 02 06:18:36 2021 +0000
@@ -62,6 +62,7 @@
 
  private:
   bool IsRDDProcessLaunching();
+  bool IsRDDProcessDestroyed() const;
   bool CreateVideoBridge();
 
   // Called from our xpcom-shutdown observer.
